### PR TITLE
MGMT-7549: extract openshift-install for none platform

### PR DESCRIPTION
**Pull request by danielerez**
# Assisted Pull Request

## Description

- Added support in Release to extract according to platform type:
  - Extract openshift-install binary for none platform.
  - Otherwise, extract openshift-baremetal-install.
- For now, non default CPU architectures (x86_64) are supported only with User Managed Networking.
- Added none to platform_type to swagger.
  - None value is used only for non x86_64 architectures (i.e. not used by SNO for now)

## List all the issues related to this PR

- [x] New Feature
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [x] Cloud
- [ ] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [x] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

The change was tested with a 4.9 release image for arm64 (not yet supported), so no real effect on behaviour, just a preparation for supporting arm64.

## Assignees

/cc @eranco74 
/cc @filanov 

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -432,6 +437,9 @@ func (b *bareMetalInventory) RegisterClusterInternal(
 		params.NewClusterParams.UserManagedNetworking = swag.Bool(true)
 		// in case of single node VipDhcpAllocation should be always false
 		params.NewClusterParams.VipDhcpAllocation = swag.Bool(false)
+	} else if cpuArchitecture != common.DefaultCPUArchitecture {
+		err = errors.Errorf("Non x86_64 CPU architectures are supported only for single node OpenShift")
```
Comment (eranco74) [inline internal/bminventory/inventory.go:441]: Really?
This is news to me.

Comment (danielerez) [reply internal/bminventory/inventory.go:441]: We can start with that until Ironic is supported for arm, and the x86 baremetal-install is available in the arm release image.

Comment (danielerez) [reply internal/bminventory/inventory.go]: Changed the validation to ensure User Managed Networking, as seems none platform multi-node can be indeed installed as well.

File: internal/bminventory/inventory.go [hunk 2]
```diff
@@ -665,6 +665,10 @@ func (b *bareMetalInventory) getNewClusterCPUArchitecture(newClusterParams *mode
 		return common.DefaultCPUArchitecture, nil
 	}
 
+	if !swag.BoolValue(newClusterParams.UserManagedNetworking) {
+		return "", errors.Errorf("Non x86_64 CPU architectures are supported only with User Managed Networking")
```
Comment (eranco74) [inline internal/bminventory/inventory.go:669]: :+1:

**Review by eranco74 — COMMENTED** [review]
/lgtm

**Pull request merged**